src/mbgl/layout/symbol_instance.hpp

```diff
@@ -28,6 +28,7 @@ class SymbolInstance {
     CollisionFeature textCollisionFeature;
     CollisionFeature iconCollisionFeature;
     WritingModeType writingModes;
+    std::size_t featureIndex;
```

Comment:
In GL JS, we did this with a direct reference to the associated feature.  Would it be better to do a similar thing here, making this a `SymbolFeature *`?